chain-abci/src/app/app_init/validator_state.rs

```diff
@@ -337,6 +337,11 @@ impl ValidatorState {
                 let mut state = get_account(addr, &root, &accounts)
                     .expect("io error or validator account not exists");
 
+                if state.is_jailed() {
+                    // don't reward jailed validator
+                    continue;
+                }
+
                 let amount = (share * count).unwrap();
```

Comment:
won't the shares/count then be incorrect?

Reply:
> won't the shares/count then be incorrect?

The share might be smaller than it should be, but if in normal case we will remove reward_stat when jailing validator, this code is only an insurance.

Reply:
log error / warn then if this shouldn't happen?

Reply:
Done